staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -458,37 +489,37 @@ func (g *Cloud) getInstancesByNames(names []string) ([]*gceInstance, error) {
 				continue
 			}
 			if found[inst.Name] != nil {
-				klog.Errorf("Instance name %q was duplicated (in zone %q and %q)", inst.Name, zone, found[inst.Name].Zone)
+				klog.Errorf("Instance name %q was duplicated (in zone %q and %q)", inst.Name, zone, found[inst.Name].instance.Zone)
 				continue
 			}
-			found[inst.Name] = &gceInstance{
-				Zone:  zone,
-				Name:  inst.Name,
-				ID:    inst.Id,
-				Disks: inst.Disks,
-				Type:  lastComponent(inst.MachineType),
+			if allNamesSet.Has(inst.Name) {
```

Comment:
why do we do this check, what happens if its not in this set, seems like at least a warning should be thrown that we seem to have listed items we don't care about

Reply:
Previously it was possible to list all instances in a project and then the function return all of that. While this is the actual behavior, from looking at the use cases of the function it is not the *intended* behavior. This is actually a bug fix to prevent that.

Reply:
ack. can you write the corresponding `else` statement and throw a warning (since that is unintended behavior)